File: py/day3-reverse-string.py
##
 # 100 Days Coding Challenge
 # 
 # @author Rhan Miano
 # @since November 6, 2018
 # 
 # Pushing myself to do this 100 Days Coding Challenge.
##

# Day 3 Reverse a string
#
# Init a string variable that will hold the reversed
# string. Set a -1 index and loop through a range of
# the startIndex up to the negative value of the length
# of the string. Third param for range is the step w/c
# the iterator should take. A value of -1 is being added,
# so that iteration would run backwards. Each args[i]
# is then concatenated to reversedStr.

def reverseString(args):
	reversedStr = ''
	startIndex  = -1
	
	for i in range(-1, -(len(args) + 1), -1):
		reversedStr = reversedStr + args[i]
		
	return reversedStr
	
print(reverseString("Hope you're doing good"))
# TODO validate if param is string

File: php/day3-reverse-string.php
<?php
/**
 * 100 Days Coding Challenge
 * 
 * @author Rhan Miano
 * @since November 6, 2018
 * 
 * Pushing myself to do this 100 Days Coding Challenge.
 */

// Day 3 Reverse a stringReverse the provided string.
//   Reverse the provided string.
//   You may need to turn the string into an array before you can reverse it.
//   The result must be string.

/**
 * Check a param if it's a valid string. If yes
 * init a variable that will hold the reversed
 * string and temp array variable. Loop through 
 * the length of args less 1 down to 0 and get the 
 * portion of $args(between current index, and a step)
 * Iteration is done backwards so pushed chars to 
 * arrTemp are from right to left. Loop through arrTemp
 * to concat each char to reversedStr.
 *
 * NOTE: Php has strrev() to do the job.
 */
function reverseString($args){
  if(is_string($args)) {
    $reversedStr = '';
    $arrTemp     = [];
    
    for($i=strlen($args)-1; $i>=0; $i--){
      $char = substr($args, $i, 1);
      array_push($arrTemp, $char);
    }
    
    foreach ($arrTemp as $char){
      $reversedStr = $reversedStr . $char;
    }
    
    return $reversedStr;
  } else {
    throw new Exception('Parameter should be a string');
  }  
}

try{
  echo reverseString("I hope that you're doing fine");
} catch(Exception $e){
  echo "Error: " . $e->getMessage();
} 